issue.go

```diff
@@ -207,3 +257,85 @@ func deleteIssueHandler(c *gin.Context, db *sql.DB) {
 	c.Status(http.StatusNoContent)
 
 }
+
+func putLocKHandler(c *gin.Context, db *sql.DB) {
+
+	_, err := authorization(c, db)
+	if err != nil {
+		return
+	}
+	username := c.Param("owner")
+	repoName := c.Param("repo")
+	issueNumber := c.Param("issue_number")
+
+	var Id int
+	row := db.QueryRow(`WITH repo_cte AS (select repos.id FROM repos JOIN users ON repos.user_id= users.id 
+                		WHERE repos.name= $1 AND users.username= $2) select id from issues where repo_id 
+                		in (select id from repo_cte) and issue_number=$3`, repoName, username, issueNumber)
+	err = row.Scan(&Id)
+	if err != nil {
+		fmt.Println(err)
+		c.AbortWithStatus(http.StatusInternalServerError)
+		return
+	}
+
+	issue := Issue{}
+	err = c.BindJSON(&issue)
+	if err != nil {
+		fmt.Println(err)
+		c.AbortWithStatus(http.StatusInternalServerError)
+		return
+	}
+
+	_, err = db.Exec(`UPDATE issues SET lock = $1 WHERE id = $2`, issue.Lock, Id)
```

Comment:
Check if the above CTE can be merged into this.

Reply:
done